fmriprep/workflows/bold/confounds.py

```diff
@@ -290,18 +303,83 @@ def _pick_wm(files):
         # ICA-AROMA
         ica_aroma_wf = init_ica_aroma_wf(name='ica_aroma_wf',
                                          ignore_aroma_err=ignore_aroma_err)
+        # performs partial regression
+        reg_filt_bold = pe.Node(fsl.FilterRegressor(), name='reg_filt_bold')
+        reg_filt_t1 = pe.Node(fsl.FilterRegressor(), name='reg_filt_t1')
+        reg_filt_mni = pe.Node(fsl.FilterRegressor(), name='reg_filt_mni')
+
+        # load csv as list for reg_filt's 'filter_columns'
+        def loadcsv(csv):
+            import numpy as np
+            return list(np.loadtxt(csv, delimiter=",", dtype='int'))
+
         workflow.connect([
             (inputnode, ica_aroma_wf, [('bold_mni', 'inputnode.bold_mni'),
                                        ('bold_mask_mni', 'inputnode.bold_mask_mni'),
                                        ('movpar_file', 'inputnode.movpar_file')]),
+            # native space
+            (ica_aroma_wf, reg_filt_bold, [('outputnode.melodic_mix', 'design_file'),
+                                           (('outputnode.aroma_noise_ics', loadcsv),
+                                           'filter_columns')]),
+            (inputnode, reg_filt_bold, [('bold', 'in_file'),
+                                        ('bold_mask', 'mask')]),
+            (reg_filt_bold, outputnode, [('out_file', 'bold_nonaggr_denoised')]),
+            # t1w space
+            (ica_aroma_wf, reg_filt_t1, [('outputnode.melodic_mix', 'design_file'),
+                                         (('outputnode.aroma_noise_ics', loadcsv),
+                                         'filter_columns')]),
+            (inputnode, reg_filt_t1, [('bold_t1', 'in_file'),
+                                      ('bold_mask_t1', 'mask')]),
+            (reg_filt_t1, outputnode, [('out_file', 'bold_nonaggr_denoised_t1')]),
+            # mni space
+            (ica_aroma_wf, reg_filt_mni, [('outputnode.melodic_mix', 'design_file'),
+                                          (('outputnode.aroma_noise_ics', loadcsv),
+                                          'filter_columns')]),
+            (inputnode, reg_filt_mni, [('bold_mni', 'in_file'),
+                                       ('bold_mask_mni', 'mask')]),
+            (reg_filt_mni, outputnode, [('out_file', 'bold_nonaggr_denoised_mni')]),
             (ica_aroma_wf, concat,
                 [('outputnode.aroma_confounds', 'aroma')]),
             (ica_aroma_wf, outputnode,
                 [('outputnode.out_report', 'ica_aroma_report'),
                  ('outputnode.aroma_noise_ics', 'aroma_noise_ics'),
                  ('outputnode.melodic_mix', 'melodic_mix'),
-                 ('outputnode.nonaggr_denoised_file', 'nonaggr_denoised_file')])
+                 ('outputnode.bold_smooth_nonaggr_denoised_mni',
+                 'bold_smooth_nonaggr_denoised_mni')]),
         ])
+
+        # replace inputnode.bold with reg_filt's out_file
+        workflow.disconnect([
```

Comment:
Why this disconnect is needed?

Reply:
This is for option 1 to avoid conflicts with the original bold being passed into these nodes (as opposed the denoised bold), alternatively, I can reconstruct the workflow so that the "correct" input is passed into the nodes from the get-go instead of deconstructing/reconstructing the workflow in case `--use-aroma` is specified.

Reply:
We generally prefer `if .. else` structures for this type of logic, as it is generally more readable.